back/app/services/permissions/idea_permissions_service.rb

```diff
@@ -1,40 +1,41 @@
-# frozen_string_literal: true
-
 module Permissions
-  class IdeaPermissionsService < ProjectPermissionsService
+  class IdeaPermissionsService < BasePermissionsService
     IDEA_DENIED_REASONS = {
       idea_not_in_current_phase: 'idea_not_in_current_phase'
     }.freeze
 
-    def denied_reason_for_action(action, user, idea, reaction_mode: nil)
-      reason = super action, user, idea.project, reaction_mode: reaction_mode
+    def initialize(idea, user, user_requirements_service: nil)
+      super(user, user_requirements_service: user_requirements_service)
+      @idea ||= idea
+    end
+
+    def denied_reason_for_idea(action, reaction_mode: nil)
```

Comment:
I renamed `denied_reason_for_action` when I saw other methods like `denied_reason_for_idea_reaction `. But now I'm thinking perhaps I should revert this and go back to ``denied_reason_for_action`, or perhaps just `denied_reason`.